lib/manager/terraform/lockfile/util.ts

```diff
@@ -0,0 +1,204 @@
+const hashLineRegex = /^(?<prefix>\s*")(?<hash>[^"]+)(?<suffix>",.*)$/;
+
+const lockFile = '.terraform.lock.hcl';
+
+export function readLockFile(): Promise<string> {
+  return readLocalFile(lockFile, 'utf8');
+}
+
+export function extractLocks(lockFileContent: string): ProviderLock[] {
+  const lines = lockFileContent.split('\n');
+  const blockStarts: number[] = [];
+  // get first lines of blocks
+  lines.forEach((line, index) => {
+    if (line.startsWith('provider "')) {
+      blockStarts.push(index);
+    }
+  });
+
+  // sort ascending
+  const sortedStarts = blockStarts.sort((a, b) => a - b);
+  const contentSlices = sortedStarts.map((start, index, array) => {
+    let end: number;
+    if (index < array.length - 1) {
+      end = array[index + 1];
+    } else {
+      end = lines.length - 1;
+    }
+    const slice: ProviderSlice = {
+      lines: lines.slice(start, end),
+      block: {
+        start,
+        end,
+      },
+    };
+    return slice;
+  });
+
+  // generate Lock objects from slices
+  const locks = contentSlices.map((slice) => {
+    let lookupName = '';
+    let registryUrl = '';
+    let version = '';
+    let constraints = '';
+    const relativeLineNumbers: LineNumbers = {
+      block: slice.block,
+      hashes: {
+        start: -1,
+        end: -1,
+      },
+    };
+    const hashes = [];
+
+    slice.lines.forEach((line, index) => {
+      const hashLineResult = hashLineRegex.exec(line);
+      if (hashLineResult) {
+        hashes.push(hashLineResult.groups.hash);
+        relativeLineNumbers.hashes.start =
+          relativeLineNumbers.hashes.start === -1
+            ? index
+            : relativeLineNumbers.hashes.start;
+        relativeLineNumbers.hashes.end = index;
+        return;
+      }
+
+      const providerStartLineResult = providerStartLineRegex.exec(line);
+      if (providerStartLineResult) {
+        lookupName = `${providerStartLineResult.groups.namespace}/${providerStartLineResult.groups.depName}`;
+        registryUrl = providerStartLineResult.groups.registryUrl;
+        return;
+      }
+
+      const versionLineResult = versionLineRegex.exec(line);
+      if (versionLineResult) {
+        version = versionLineResult.groups.version;
+        relativeLineNumbers.version = index;
+        return;
+      }
+
+      const constraintLineResult = constraintLineRegex.exec(line);
+      if (constraintLineResult) {
+        constraints = constraintLineResult.groups.constraint;
+        relativeLineNumbers.constraint = index;
+      }
+    });
+
+    const lock: ProviderLock = {
+      lookupName,
+      registryUrl,
+      version,
+      constraints,
+      hashes,
+      lineNumbers: relativeLineNumbers,
+    };
+    return lock;
+  });
+
+  if (locks.length === 0) {
+    return null;
+  }
+  return locks;
+}
+
+export function isPinnedVersion(value: string): boolean {
+  const versioning = getVersioning('hashicorp');
+  return <boolean>versioning.isSingleVersion(value);
```

Comment:
```suggestion
  return !!versioning.isSingleVersion(value);
```

do not use `<boolean>` cast use double negation or  `as boolean` cast.